sound/soc/sof/intel/hda-pcm.c

```diff
@@ -269,6 +269,6 @@ int hda_dsp_pcm_close(struct snd_sof_dev *sdev,
 	}
 
 	/* unbinding pcm substream to hda stream */
-	substream->runtime->private_data = NULL;
+	substream->private_data = NULL;
```

Comment:
That's really not so good, if you really want to have a cleaner patch you should not directly handle the private_data but use helpers to store.

Actually you should first introduce the helpers to remove all directly handling of private_data, and you have the location in one shot.